File: app/src/main/java/com/jampez/sidebysideexample/MainActivity.java
package com.jampez.sidebysideexample;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import com.jampez.sidebyside.SideBySideView;

public class MainActivity extends AppCompatActivity {

    SideBySideView names, contactInfo, passwords, authCode;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final TextView text = findViewById(R.id.text);
        names = findViewById(R.id.names);
        names.setLeftInput("testing");
        contactInfo = findViewById(R.id.contact_info);
        passwords = findViewById(R.id.password);
        authCode = findViewById(R.id.authorisation_code);
        names.setLeftInputListener(new TextWatcher() {
            @Override public void beforeTextChanged(CharSequence s, int start, int count, int after) { }
            @Override public void onTextChanged(CharSequence s, int start, int before, int count) { }
            @Override public void afterTextChanged(Editable s) {
                text.setText(s.toString());
            }
        });

        Button validate = findViewById(R.id.validate);
        validate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                boolean namesValid = names.isValid();
                boolean contactInfoValid = contactInfo.isValid();
                boolean passwordsValid = passwords.isValid();
                boolean authCodeValid = authCode.isValid();

                if(namesValid && contactInfoValid && passwordsValid && authCodeValid)
                    text.setText("Validated!");
                else
                    text.setText("Not Validated!");
            }
        });
    }

    @Override
    public void onBackPressed(){
        if(!names.haveInputsBeenEdited())
            finish();
        else {
            AlertDialog.Builder alertDialog = new AlertDialog.Builder(this);
            alertDialog.setTitle("Are you sure you want to quit?");
            alertDialog.setCancelable(false);
            alertDialog.setNegativeButton("No", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    dialog.dismiss();
                }
            });
            alertDialog.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    finish();
                }
            });
            alertDialog.create();
            alertDialog.show();
        }
    }
}

File: app/src/main/java/com/jampez/sidebysideexample/MainActivity.kt
package com.jampez.sidebysideexample

import android.annotation.SuppressLint
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.jampez.sidebyside.SideBySideView

class MainActivity : AppCompatActivity() {
    private lateinit var names: SideBySideView
    private lateinit var contactInfo: SideBySideView
    private lateinit var passwords: SideBySideView
    private lateinit var authCode: SideBySideView
    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val text = findViewById<TextView>(R.id.text)
        names = findViewById(R.id.names)
        contactInfo = findViewById(R.id.contact_info)
        passwords = findViewById(R.id.password)
        authCode = findViewById(R.id.authorisation_code)
        names.setLeftInputListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable) {
                text.text = s.toString()
            }
        })
        val validate = findViewById<Button>(R.id.validate)
        validate.setOnClickListener {
            val namesValid = names.isValid
            val contactInfoValid = contactInfo.isValid
            val passwordsValid = passwords.isValid
            val authCodeValid = authCode.isValid
            if (namesValid && contactInfoValid && passwordsValid && authCodeValid) text.text = "Validated!" else text.text = "Not Validated!"
        }
    }

    override fun onBackPressed() {
        if (!names.haveInputsBeenEdited()) finish() else {
            val alertDialog = AlertDialog.Builder(this)
            alertDialog.setTitle("Are you sure you want to quit?")
            alertDialog.setCancelable(false)
            alertDialog.setNegativeButton("No") { dialog, _ -> dialog.dismiss() }
            alertDialog.setPositiveButton("Yes") { _, _ -> finish() }
            alertDialog.create()
            alertDialog.show()
        }
    }
}